framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileElementTypes.java

```diff
@@ -628,15 +627,23 @@ private void parseEnclosingClass(Element e) {
             return;
         }
         String className = getOutermostEnclosingClass(e);
-        if (className == null || className.isEmpty()) {
+        if (className == null || className.isEmpty() || processingClasses.contains(className)) {
             return;
         }
-        if (jdkStubFiles.containsKey(className)) {
-            parseJdkStubFile(jdkStubFiles.get(className));
-            jdkStubFiles.remove(className);
-        } else if (jdkStubFilesJar.containsKey(className)) {
-            parseJdkJarEntry(jdkStubFilesJar.get(className));
-            jdkStubFilesJar.remove(className);
+
+        boolean prevParsing = parsing;
```

Comment:
This pattern is generally safer. Is it worth cleaning up the other places in this class that simply set `parsing` to true/false?

Reply:
Thanks! I have changed the boolean to a counter.